### PR TITLE
Apriltag pose estimator

### LINKED ISSUE #5: Add AprilTag pose estimation
Add this all in the Vision subsystem (#2)
Kinda just follow this: https://docs.photonvision.org/en/latest/docs/programming/photonlib/robot-pose-estimator.html

- [x] Add static IP to the beelink
- [x] Update PhotonVision to v2023.1.2 
- [x] Create Photoncamera(s) (just start with 1 for now)
- [x] Create robotToCamera transform
- [x] Create RobotPoseEstimator (from PhotonVision, not WPI) 
  - using the CLOSEST_TO_REFERENCE_POSE strategy 
  - Name this something like "photonPoseEstimator" to avoid confusion with the WPI pose estimator
- [x] Add getEstimatedGlobalPose() 
  - Again, probably rename this to avoid confusion
- [x] Implement getEstimatedGlobalPose() with WPI pose estimator (relies on #1 or whatever issue creates the WPI pose estimator)
- [x] Yippee 🎉

### PR DESCRIPTION
**Pull request by ACat701**
Closes #5

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Vision.java [hunk 1]
```diff
@@ -4,10 +4,55 @@
...
 
 public class Vision extends SubsystemBase {
+  RobotPoseEstimator photonPoseEstimator;
+
   public Vision() {
+    // TODO: Change to current april tag array upon new WPI update
+    AprilTagFieldLayout aprilTagFieldLayout = new AprilTagFieldLayout(null, 0, 0);
+
+    // add more cameras here & to the cameraList when possible
+    PhotonCamera lifecam = new PhotonCamera("Microsoft_LifeCam_HD-3000");
```
Comment (IanSzalai) [inline src/main/java/frc/robot/subsystems/Vision.java:36]: either camera names or PhotonCamera objects should probably be in constants

File: src/main/java/frc/robot/subsystems/Vision.java [hunk 2]
```diff
@@ -4,10 +4,55 @@
...
+
   public Vision() {
+    try {
+      aprilTagFieldLayout = new AprilTagFieldLayout(AprilTagFields.k2023ChargedUp.m_resourceFile);
+    } catch (Exception e) {
+      System.out.println("Could not load AprilTagFieldLayout! Must be on WPILIB 2023.2.1+" + e);
+    }
+
+    PhotonCamera ARCamera = new PhotonCamera(constVision.ARPhotonName);
+    Transform3d robotToAR = new Transform3d(new Translation3d(0, 0, 0), new Rotation3d(0, 0, 0));
```
Comment (IanSzalai) [inline src/main/java/frc/robot/subsystems/Vision.java:38]: should be a constant

File: src/main/java/frc/robot/subsystems/Vision.java [hunk 3]
```diff
@@ -4,10 +4,55 @@
...
+
+    PhotonCamera OVCamera = new PhotonCamera(constVision.OVPhotonName);
+    Transform3d robotToOV = new Transform3d(new Translation3d(0, 0, 0), new Rotation3d(0, 0, 0));
```
Comment (IanSzalai) [inline src/main/java/frc/robot/subsystems/Vision.java:41]: also should be a constant

**Comment by IanSzalai** [comment]
this is probably ready for review, @ACat701 you think so?

**Comment by ACat701** [comment]
> this is probably ready for review, @ACat701 you think so?

Yeah. If we want to change the pose estimation strategy in the future that can be a separate issue (but I don't think we're doing that anyway)

**Review by AngelaKimmy — DISMISSED** [review]
Looks good to me. Nice and clean too! :)

File: src/main/java/frc/robot/Constants.java [hunk 4]
```diff
@@ -6,17 +6,32 @@
...
+    public static final Transform3d robotToOV = new Transform3d(new Translation3d(0, 0, 0), new Rotation3d(0, 0, 0));
+    public static final Transform3d robotToLifecam = new Transform3d(new Translation3d(0.4191, -0.1905, 0.6604),
+        new Rotation3d(0, 0, 0));
+  }
+
   public enum GamePiece {
     NONE, CUBE, CONE
   }
 
-  public static final boolean OUTPUT_DEBUG_VALUES = false;
```
Comment (TaylerUva) [inline src/main/java/frc/robot/Constants.java:19]: This should be set back to false

**Review by TaylerUva — CHANGES_REQUESTED** [review]

**Review by TaylerUva — APPROVED** [review]

**Pull request merged**